doc/sphinx/proof-engine/vernacular-commands.rst

```diff
@@ -568,67 +493,50 @@ Compiled files
+   Loads compiled modules into the |Coq| environment.  The command
+   searches the loadpath for the modules named by the :n:`@dirpath`\s.  Each :n:`@dirpath`
+   has the form :n:`{* @ident .}@ident`.  The command maps the prefix :n:`{* @ident .}` to a physical
+   directory (see Section :ref:`libraries-and-filesystem`) and using the final :n:`@ident`,
+   loads the compiled
+   file :n:`@ident.vo` from that directory.  (Compiling :n:`@ident.v` generates :n:`@ident.vo`.)
 
+   The process is applied recursively to all the loaded modules.
+   The compiled files must have been compiled with the same version of |Coq|.
+   The compiled files are neither replayed nor rechecked.
 
-   .. cmdv:: Require Import @qualid
-      :name: Require Import
+   * :n:`Import` - additionally does an :cmd:`Import` on the loaded module, making components defined
+     in the module available by their short names
+   * :n:`Export` - additionally does an :cmd:`Export` on the loaded module, making components defined
+     in the module available by their short names *and* marking them to be exported by the current
+     module
 
-      This loads and declares the module :n:`@qualid`
-      and its dependencies then imports the contents of :n:`@qualid` as described
-      for :cmd:`Import`. It does not import the modules that
-      :n:`@qualid` depends on unless these modules were themselves required in module
-      :n:`@qualid`
-      using :cmd:`Require Export`, or recursively required
-      through a series of :cmd:`Require Export`.  If the module required has
-      already been loaded, :cmd:`Require Import` :n:`@qualid` simply imports it, as
-      :cmd:`Import` :n:`@qualid` would.
+   If the required module has already been loaded, :n:`Import` and :n:`Export` make the command
+   equivalent to :cmd:`Import` or :cmd:`Export`.
 
-   .. cmdv:: Require Export @qualid
-      :name: Require Export
-
-      This command acts as :cmd:`Require Import` :n:`@qualid`,
-      but if a further module, say `A`, contains a command :cmd:`Require Export` `B`,
-      then the command :cmd:`Require Import` `A` also imports the module `B.`
+   The mapping between
+   physical directories and logical names at the time of requiring the
+   file must be consistent with the mapping used to compile the file. If
+   several files match, one of them is picked in an unspecified fashion.
 
-   .. cmdv:: Require {| Import | Export } {+ @qualid }
+   .. todo common user error on dirpaths see https://github.com/coq/coq/pull/11961#discussion_r402852390
 
-      This loads the
-      modules named by the :token:`qualid` sequence and their recursive
-      dependencies. If
-      ``Import`` or ``Export`` is given, it also imports these modules and
-      all the recursive dependencies that were marked or transitively marked
-      as ``Export``.
+.. cmd:: From @qualid Require {? {| Import | Export } } {+ @qualid }
+   :name: From ... Require ...
```

Comment:
I think `...` should be used to denote holes only in the middle of an unfinished command.

```suggestion
   :name: From ... Require
```